### PR TITLE
Validation of Widget's source id field

### PR DESCRIPTION
**Pull request by mayoral**
This PR implements #12878.

### CODE REVIEW

File: app/models/carto/widget.rb [hunk 1]
```diff
@@ -88,6 +89,13 @@ def validate_user_not_viewer
     end
   end
 
+  def valid_source_id
+    if source_id.present? && !source_id.is_a?(String)
```
Comment (juanignaciosl) [inline app/models/carto/widget.rb]: Can `source_id` be `nil` or empty`? Maybe we can add a presence validation as well.

File: app/models/carto/widget.rb [hunk 2]
```diff
@@ -88,6 +89,13 @@ def validate_user_not_viewer
...
+      errors.add(:source_id, "Source id must be a string")
+      false
```
Comment (juanignaciosl) [inline app/models/carto/widget.rb]: [According to the docs](http://guides.rubyonrails.org/v3.2.20/active_record_validations_callbacks.html#custom-methods) there's no need to return `false`, is it?

File: app/models/carto/widget.rb (hunk 1 above)
Comment (mayoral) [reply app/models/carto/widget.rb]: 👍 . Added presence validation in 726a89e.

File: app/models/carto/widget.rb (hunk 2 above)
Comment (mayoral) [reply app/models/carto/widget.rb]: Yep! I usually don't add this false and only add the error to the array, but as I saw in the previous validation in the file the `false` I thought that maybe custom validation had a behaviour similar to the callbacks that can stop execution returning false, but I just did a quick check with code and adding false doesn't stop the execution chain when calling `valid?`, so I've removed my usage and the usage in the other validator.

Comment (juanignaciosl) [reply app/models/carto/widget.rb]: Yep, you might find some validations returning a boolean at methods that are also useful as public, "normal" methods. Nevertheless, it's not usually needed and it's therefore discouraged.

**Comment by amartinj** [comment]
I've been playing around creating several kinds of widgets, adding them to the map, sharing it, accessing it and everything works fine. Moving to deploy

**Pull request merged**